app/controllers/applications_controller.rb

```diff
@@ -0,0 +1,44 @@
+class ApplicationsController < ApplicationController
+  def show
+    @application = Application.find(params[:id])
+    if params[:search_app].present?
+      @pets = Pet.search(params[:search_app])
+    end
+
+    if params[:pet_id].present?
+      @application.add_pet(params[:pet_id])  
+    end
+  end
+
+  def new
+  end
+
+  def create
+    admin = Admin.create
+    @application = admin.applications.create({
+      name: params[:name],
+      address: params[:address],
+      city: params[:city],
+      state: params[:state],
+      zipcode: params[:zipcode],
+      description: params[:description],
+      status: "In Progress"
+    })
+    if @application.valid?
+      redirect_to "/applications/#{@application.id}"
+    else 
+      @application.save
```

Comment:
Instead of doing `.valid` and `.save` in two different steps, you can do it at once and just use `.save` (which will run .valid under the hood)
```
@application = Application.new(params...)
if @application.save
    happy_path_stuff
else
   flash errors by accessing `@application.errors.full_messages`
end